code/game/machinery/cloning.dm

```diff
@@ -348,10 +346,66 @@
 			domutcheck(occupant) //Waiting until they're out before possible transforming.
 	occupant = null
 
-	biomass -= CLONE_BIOMASS
 	update_icon()
 	return
 
+// Returns the total amount of biomass reagent in all of the pod's stored containers
+/obj/machinery/clonepod/proc/get_biomass()
+	var/biomass_count = 0
+	if(LAZYLEN(containers))
+		for(var/obj/item/weapon/reagent_containers/glass/G in containers)
+			for(var/datum/reagent/R in G.reagents.reagent_list)
+				if(R.id == "biomass")
+					biomass_count += R.volume
+
+	return biomass_count
+
+// Removes [amount] biomass, spread across all containers. Doesn't have any check that you actually HAVE enough biomass, though.
+/obj/machinery/clonepod/proc/remove_biomass(var/amount = CLONE_BIOMASS)		//Just in case it doesn't get passed a new amount, assume one clone
+	var/to_remove = 0	// Tracks how much biomass has been found so far
+	if(LAZYLEN(containers))
+		for(var/obj/item/weapon/reagent_containers/glass/G in containers)
+			if(to_remove < amount)	//If we have what we need, we can stop. Checked every time we switch beakers

```

Comment:
Does this not always return 1, as to_remove starts out as 0, less than the amount needed? If to_remove is how much has already been found, it would make sense if it checked it for being greater than what has been found.

Reply:
Look at what the line below does. If the machine hasn't found enough biomass, it runs through the beakers and removes 60 total.
If 'to_remove >= amount', then we just return, because we've removed enough.

Reply:
..Yes I'm just an idiot and now question why I'm doing this tired.